cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/ElasticsearchSink.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.batch.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.api.templates.plugins.PluginConfig;
+import co.cask.cdap.template.etl.api.Emitter;
+import co.cask.cdap.template.etl.api.batch.BatchSink;
+import co.cask.cdap.template.etl.api.batch.BatchSinkContext;
+import co.cask.cdap.template.etl.common.Properties;
+import co.cask.cdap.template.etl.common.RecordUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.Job;
+import org.elasticsearch.hadoop.mr.EsOutputFormat;
+
+import javax.annotation.Nullable;
+
+/**
+ * An ElasticsearchSink
+ */
+@Plugin(type = "sink")
+@Name("Elasticsearch")
+@Description("CDAP Elasticsearch Batch Sink")
+public class ElasticsearchSink extends BatchSink<StructuredRecord, Writable, Writable> {
+  private static final String INDEX_DESC = "The name of the index where the data will be stored. " +
+    "The index should already exist and be configured.";
+
+  private static final String TYPE_DESC = "The name of the type where the data will be stored";
+
+  private static final String ID_DESC = "The id field that will determine the id for the document. " +
+    "It should match a fieldname in the structured record of the input";
+
+  private static final String HOST_DESC = "The hostname and port for an elasticsearch node. e.g. localhost:9300";
+
+  /**
+   * Config class for RealtimeTableSink
+   */
+  public static class ESConfig extends PluginConfig {
```

Comment:
also can we move this class to the bottom of the class? it's sandwiched awkwardly between final variables